components/site/src/link_checking.rs

```diff
@@ -47,23 +47,26 @@ pub fn check_internal_links_with_anchors(site: &Site) -> Result<()> {
 
     // Check for targets existence (including anchors), then keep only the faulty
     // entries for error reporting purposes.
-    let missing_targets = links_with_anchors.filter(|(_, md_path, anchor)| {
+    let missing_targets = links_with_anchors.filter(|(page, md_path, anchor)| {
         // There are a few `expect` here since the presence of the .md file will
         // already have been checked in the markdown rendering
         let mut full_path = site.base_path.clone();
         full_path.push("content");
         for part in md_path.split('/') {
             full_path.push(part);
         }
-        if md_path.contains("_index.md") {
+        // NOTE: This will also match _index.foobar.md where foobar is not a language
+        // I don't think this is supported by zola so nothing should be broken
+        // Will also prevent having a section like content/_index.foobar/_index.md
+        if md_path.contains("/_index.") {
```

Comment:
Does the root section starts with `/`? I can't remember

Reply:
Very good point i removed the `/` as initially suggested on #1694. I added a note in the docs that any **page** whose path contains `_index.` is unsupported.